snapcraft/plugins/_python/_pip.py

```diff
@@ -0,0 +1,339 @@
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode(sys.getfilesystemencoding()).strip()
+            if 'no module named pip' in output.lower():
+                return False
+            else:
+                raise e
+
+        return True
+
+    def download(self, packages, *, setup_py_dir=None, constraints=None,
+                 requirements=None, process_dependency_links=False):
+        """Download packages into cache, but do not install them.
+
+        :param iterable packages: Packages to download from index.
+        :param str setup_py_dir: Directory containing setup.py.
+        :param iterable constraints: Collection of paths to constraints files.
+        :param iterable requirements: Collection of paths to requirements
+                                      files.
+        :param boolean process_dependency_links: Enable the processing of
+                                                 dependency links.
+        """
+        args = []
+        cwd = None
+        if setup_py_dir:
+            args.append('.')
+            cwd = setup_py_dir
+
+        _process_common_args(
+            args, process_dependency_links=process_dependency_links,
+            packages=packages, constraints=constraints,
+            requirements=requirements)
+
+        if not args:
+            return  # No operation was requested
+
+        # Using pip with a few special parameters:
+        #
+        # --disable-pip-version-check: Don't whine if pip is out-of-date with
+        #                              the version on pypi.
+        # --dest: Download packages into the directory we've set aside for it.
+        self._run(['download', '--disable-pip-version-check', '--dest',
+                   self._python_package_dir] + args, cwd=cwd)
+
+    def install(self, packages, *, setup_py_dir=None, constraints=None,
+                requirements=None, process_dependency_links=False,
+                upgrade=False, install_deps=True, ignore_installed=False):
+        """Install packages from cache.
+
+        The packages should have already been downloaded via `download()`.
+
+        :param iterable packages: Packages to install from cache.
+        :param str setup_py_dir: Directory containing setup.py.
+        :param iterable constraints: Collection of paths to constraints files.
+        :param iterable requirements: Collection of paths to requirements
+                                      files.
+        :param boolean process_dependency_links: Enable the processing of
+                                                 dependency links.
+        :param boolean upgrade: Recursively upgrade packages.
+        :param boolean install_deps: Install package dependencies.
+        :param boolean ignore_installed: Reinstall packages if they're already
+                                         installed
+        """
+        args = []
+        cwd = None
+        if setup_py_dir:
+            args.append('.')
```

Comment:
this should be the last arg so that `python-packages` and `requirements.txt` can potentially satisfy `install_requires`

Reply:
Ah ha, good to know! Thank you :) .

Reply:
Done.